internal/quickfort/query.lua

```diff
@@ -5,12 +5,117 @@ if not dfhack_flags.module then
     qerror('this script cannot be called directly')
 end
 
+local gui = require('gui')
+local guidm = require('gui.dwarfmode')
 local quickfort_common = reqscript('internal/quickfort/common')
 local log = quickfort_common.log
+local quickfort_aliases = reqscript('internal/quickfort/aliases')
+local quickfort_keycodes = reqscript('internal/quickfort/keycodes')
+
+local common_aliases_filename = 'hack/data/quickfort/aliases-common.txt'
+local user_aliases_filename = 'dfhack-config/quickfort/aliases.txt'
+
+local function load_aliases()
+    -- ensure we're starting from a clean alias stack, even if the previous
+    -- invocation of this function returned early with an error
+    quickfort_aliases.reset_aliases()
+    quickfort_aliases.push_aliases_csv_file(common_aliases_filename)
+    quickfort_aliases.push_aliases_csv_file(user_aliases_filename)
+end
+
+local function is_queryable_tile(pos)
+    local flags, occupancy = dfhack.maps.getTileFlags(pos)
+    return not flags.hidden and occupancy.building ~= 0
+end
+
+local function move_cursor(screen, overlay, pos)
+    overlay:moveCursorTo(pos)
+    -- wiggle the cursor so the building under the cursor gets properly selected
+    -- is there a better way to do this? I've tried:
+    -- - DwarfOverlay:simulateCursorMovement
+    -- - DwarfOverlay:selectBuilding
+    -- - setting the ui.main.mode to something else then back to QueryBuilding
```

Comment:
`dfhack.gui.refreshSidebar()` does pretty much exactly this (it uses the z axis, for no real reason other than the plugins that needed this before it was a library function did it that way).

Reply:
looking at the library code, that's hilarious we came up with the same "solution" : )  changed to use library version.